packages/react/src/transactions/adapters/PublicationCacheManager.ts

```diff
@@ -0,0 +1,145 @@
+import { ApolloCache, DocumentNode, NormalizedCacheObject } from '@apollo/client';
+import {
+  CommentFragmentDoc,
+  JustTypename,
+  MirrorFragmentDoc,
+  PostFragmentDoc,
+  PublicationFragment,
+} from '@lens-protocol/api-bindings';
+import { failure, invariant, never, Result, success } from '@lens-protocol/shared-kernel';
+
+export class FragmentNotFoundError extends Error {
+  name = 'FragmentNotFoundError' as const;
+}
+
+export const getFragmentDoc = (publication: JustTypename<PublicationFragment>): DocumentNode => {
+  switch (publication.__typename) {
+    case 'Mirror':
+      return MirrorFragmentDoc;
+    case 'Post':
+      return PostFragmentDoc;
+    case 'Comment':
+      return CommentFragmentDoc;
+    default:
+      never("Can't infer fragment document");
+  }
```

Comment:
I know this is not much code, but wouldn't an error in this function yield green tests. Isn't this approach failing the point of using tests as double-entry bookkeeping (the test verify the code and the code verify the test)?

Reply:
(I am more interested in the conversation than this specific piece of code)

Reply:
Can't justify my decision more than just having less code in spec to write. 
Reverting back to using cache directly in the test file.